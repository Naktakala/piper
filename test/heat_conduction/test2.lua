nodes = {}
N = 100
xmin = 0.0
L = 1.0
nodes = {}

dx = L / N
for i = 0, N do
  nodes[i + 1] = xmin + dx * i
end
meshgen1 = chi_mesh.OrthogonalMeshGenerator.Create({ node_sets = { nodes, nodes } })
chi_mesh.MeshGenerator.Execute(meshgen1)

system1 = chi_math.FEMKernelSystem.Create
({
  fields = {
    chi_physics.FieldFunctionGridBased.Create({
      name = "T",
      sdm_type = "PWLC"
    })
  },
  kernels = {
    { type = hcm.ThermalConductionKernel.type, var="T", k = 16.0 },
    --chi_math.SinkSourceFEMKernel.Create({ value = 100.0e2 })
  },
  bcs = {
    { type = chi_math.FEMDirichletBC.type, var="T", boundaries = { "XMIN" }, bc_value = -1.0 },
    { type = chi_math.FEMDirichletBC.type, var="T", boundaries = { "XMAX" }, bc_value = 10.0 },
    { type = chi_math.FEMDirichletBC.type, var="T", boundaries = { "YMIN" }, bc_value = 20.0 },
    { type = chi_math.FEMDirichletBC.type, var="T", boundaries = { "YMAX" }, bc_value = 30.0 }
  }
  --verbosity = 2
})

phys1 = chi_math.SteadyNonLinearExecutioner.Create
({
  name = "phys1",
  system = system1,
  solver_params =
  {
    nl_method = "PJFNK",
    l_rel_tol = 1.0e-5,
  }
})

chiSolverInitialize(phys1)
chiSolverExecute(phys1)

--############################################### PostProcessors
chi.CellVolumeIntegralPostProcessor.Create
({
  name = "avgval",
  field_function = "T",
  compute_volume_average = true
})
chi.ExecutePostProcessors({"avgval"})

if (master_export == nil) then
  chiExportMultiFieldFunctionToVTK({ "T" }, "test2")
end