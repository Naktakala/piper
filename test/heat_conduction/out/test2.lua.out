mpiexec -np 2 /Users/vermjc/Repos/ChiTech/piper/bin/piper test2.lua --suppress_color --supress_beg_end_timelog master_export=false 
[0]  Parsing argument 1 test2.lua
[0m[0]  Parsing argument 2 --suppress_color
[0m[0]  Parsing argument 3 --supress_beg_end_timelog
[0]  Parsing argument 4 master_export=false
[0]  ChiTech number of arguments supplied: 4
[0]  Computing cell-centroids.
[0]  Done computing cell-centroids.
[0]  Checking cell-center-to-face orientations
[0]  Done checking cell-center-to-face orientations
[0]  00:00:00 Establishing cell connectivity.
[0]  00:00:00 Vertex cell subscriptions complete.
[0]  00:00:00 Surpassing cell 1000 of 10000 (10%)
[0]  00:00:00 Surpassing cell 2000 of 10000 (20%)
[0]  00:00:00 Surpassing cell 3001 of 10000 (30%)
[0]  00:00:00 Surpassing cell 4000 of 10000 (40%)
[0]  00:00:00 Surpassing cell 5000 of 10000 (50%)
[0]  00:00:00 Surpassing cell 6001 of 10000 (60%)
[0]  00:00:00 Surpassing cell 7001 of 10000 (70%)
[0]  00:00:00 Surpassing cell 8000 of 10000 (80%)
[0]  00:00:00 Surpassing cell 9000 of 10000 (90%)
[0]  00:00:00 Surpassing cell 10000 of 10000 (100%)
[0]  00:00:00 Establishing cell boundary connectivity.
[0]  00:00:00 Done establishing cell connectivity.
[0]  MeshGenerator: Cells created = 10000
[0]  
[0]  Executing solver "phys1"
[0]  NonLinearSolver non-linear iteration    0 - Residual 5.9594651e+03
[0]  NonLinearSolver iteration    0 - Residual 5.9594651e+03
[0]  NonLinearSolver iteration    1 - Residual 7.7267776e+01
[0]  NonLinearSolver iteration    2 - Residual 4.3680907e+00
[0]  NonLinearSolver iteration    3 - Residual 4.9377014e-01
[0]  NonLinearSolver iteration    4 - Residual 4.9368631e-02
[0]  NonLinearSolver non-linear iteration    1 - Residual 4.9461155e-02
[0]  NonLinearSolver iteration    0 - Residual 4.9461155e-02
[0]  NonLinearSolver iteration    1 - Residual 5.5454044e-03
[0]  NonLinearSolver iteration    2 - Residual 4.2471054e-04
[0]  NonLinearSolver iteration    3 - Residual 2.0421506e-05
[0]  NonLinearSolver iteration    4 - Residual 1.0166014e-06
[0]  NonLinearSolver iteration    5 - Residual 1.0138915e-07
[0]  NonLinearSolver non-linear iteration    2 - Residual 1.4604598e-06
[0]            Converged CONVERGED_FNORM_RELATIVE
[0]  
[0]  
[0]  SCALAR post-processors history at event "ProgramExecuted"
[0]  *--------*-----------*
[0]  | Time   | avgval    |
[0]  *--------*-----------*
[0]  | Latest | 14.750000 |
[0]  *--------*-----------*
[0]  
[0]  SCALAR post-processors latest values at event "ProgramExecuted"
[0]  *----------------------*-----------------*
[0]  | Post-Processor Name  | Value           |
[0]  *----------------------*-----------------*
[0]  | avgval(latest)       |       14.750000 |
[0]  *----------------------*-----------------*
[0]  


